expression/builtin_encryption.go

```diff
@@ -494,12 +494,17 @@ type sha2FunctionClass struct {
 }
 
 func (c *sha2FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinSHA2Sig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	types.SetBinChsClnFlag(bf.tp)
```

Comment:
Hi, I think I need help to get hash length in getFunction.
I tried to get it from args, but failed.
``` go
func (c *sha2FunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {                                                
    bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt)
    if err != nil {
        return nil, errors.Trace(err)
    }
    bfArgs := bf.getArgs()
    hashLength, isNull, err := bfArgs[1].EvalInt(nil, ctx.GetSessionVars().StmtCtx)
    if isNull || err != nil {
        return nil, errors.Trace(err)
    }
    switch int(hashLength) {
    case SHA0:
        bf.tp.Flen = SHA256 / 4
    case SHA224, SHA256, SHA384, SHA512:
        bf.tp.Flen = int(hashLength) / 4
    default:
        return nil, errors.Trace(err)
    }
    sig := &builtinSHA2Sig{baseStringBuiltinFunc{bf}}
    return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
}
```